### PR TITLE
Add Google Summer of Code page

### PR DESCRIPTION
**Pull request by nikhita**
Add Google Summer of Code page.

xref https://github.com/cncf/soc/issues/22.

/assign parispittman idvoretskyi

### CODE REVIEW

**Comment by nikhita** [comment]
fyi @sttts

**Comment by sttts** [comment]
:+1:

**Comment by stp-ip** [comment]
Awesome. LGTM from my side. Just one small thing. There are a lot of long single lines, which could be split up into multi lines. Also a few long single line with a strange split to multi line such as: https://github.com/kubernetes/community/pull/1616/files#diff-9c2fe8e562760c0deac8021c7f8a6581R22

File: mentoring/google-summer-of-code.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+## Communication
+
+- Mailing List: development announcements and discussions appear on the Google group `kubernetes-dev` (send mail to kubernetes-dev@googlegroups.com).
+- Slack: Please [sign up](http://slack.k8s.io/) for an invite.
+
+## Contribute
+
+As part of the application process, the Google Summer of Code program recommends that candidates make small contributions to the project they intend to apply for.
+To start working on the project, make sure to fill out the [CLA](../CLA.md  ) and check if you have the right environment with this guide. The README in the [community repo](https://github.com/kubernetes/community) details these things and more.
```
Comment (stp-ip) [inline mentoring/google-summer-of-code.md]: A CLA link such as [CLA](/CLA.md) might make the file itself more movable.

Comment (nikhita) [reply mentoring/google-summer-of-code.md]: @stp-ip updated. ptal, thanks.

**Comment by stp-ip** [comment]
/lgtm
@idvoretskyi approval?

**Pull request merged**